pkg/deviceplugins/usb_device_plugin_helper.go

```diff
@@ -0,0 +1,208 @@
+package deviceplugins
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+func parseSysUeventFile(path string) *USBDevice {
+	link, err := os.Readlink(path)
+	if err != nil {
+		return nil
+	}
+
+	// Grab all details we are interested from uevent
+	file, err := os.Open(filepath.Join(path, "uevent"))
+	if err != nil {
+		logrus.Printf("Unable to access %s/%s\n", path, "uevent")
+		return nil
+	}
+	defer file.Close()
+
+	u := USBDevice{
+		PCIAddress: parseUSBSymLinkToPCIAddress(link),
+	}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		values := strings.Split(line, "=")
+		if len(values) != 2 {
+			logrus.Printf("Skipping %s due not being key=value\n", line)
+			continue
+		}
+
+		key, value := values[0], values[1]
+		if !parseSysUeventKeyValue(key, value, &u) {
+			return nil
+		}
+	}
+
+	return &u
+}
+
+func parseSysUeventKeyValue(key string, value string, u *USBDevice) bool {
+	switch key {
+	case "BUSNUM":
+		val, err := strconv.ParseInt(value, 10, 32)
```

Comment:
we are skipping return usbdevice objects if there is an error for any reason. It makes sense to ignore the error but should we log the error atleast to help debug in case of edge cases?

Reply:
sure, make sense, i'll add log